eva-server/src/main/java/uk/ac/ebi/eva/server/ws/RegionWSServerV2.java

```diff
@@ -137,7 +151,29 @@ public ResponseEntity getVariantsByRegion(@PathVariable("regionId") String regio
 
             resourcesList.add(new Resource<>(variant, Arrays.asList(sourcesLink, annotationsLink)));
         });
-        return new ResponseEntity(new Resources<>(resourcesList), HttpStatus.OK);
+
+        PagedResources pagedResources = new PagedResources<>(resourcesList, pageMetadata);
```

Comment:
I would extract the whole construction of the `PagedResources` object to an auxiliary `buildPage` method. In fact, I think this whole method could be split in 3 clear parts, and each of them becoming one:

1. Build page metadata
2. Get resources
3. Build page

Reply:
Fixed :)